packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -661,39 +765,175 @@ abstract class TableBase extends Resource implements ITable {
+      throw new Error("Both 'Operation' and 'TableName' dimensions must be passed for the 'SuccessfulRequestLatency' metric.");
+    }
+
     return this.metric('SuccessfulRequestLatency', { statistic: 'avg', ...props });
   }
 
+  /**
+   * Metric for the system errors this table.
+   *
+   * This will sum errors across all possible operations.
+   * Note that by default, each individual metric will be calculated as a sum over a period of 5 minutes.
+   * You can customize this by using the `statistic` and `period` properties.
+   */
+  public metricSystemErrorsForOperations(props?: SystemErrorsForOperationsMetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation) {
+      throw new Error("The Operation dimension is not supported. Use the 'operations' property.");
+    }
+
+    const operations = props?.operations ?? Object.values(Operation);
+
+    const values = this.createMetricsForOperations('SystemErrors', operations, { statistic: 'sum', ...props });
+
+    const sum = new cloudwatch.MathExpression({
+      expression: `${Object.keys(values).join(' + ')}`,
+      usingMetrics: { ...values },
+      color: props?.color,
+      label: props?.label ?? 'Sum over operations',
+      period: props?.period,
+    });
+
+    return sum;
+  }
+
+  /**
+   * Metric for the successful request latency for specific operations of this table.
+   *
+   * This will perform a weighted average across all possible operations.
+   * Note that by default, each individual metric will be calculated as an average over a period of 5 minutes.
+   * You can customize this by using the `statistic` and `period` properties.
+   */
+  public metricSuccessfulRequestLatencyForOperations(props: SuccessfulRequestLatencyForOperationsMetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation) {
+      throw new Error("The Operation dimension is not supported. Use the 'operations' property.");
+    }
+
+    if (props.operations.length === 0) {
+      // you must pass at least 1 operation since this is an operation dimensionality metric.
+      throw new Error("You must pass at least one operation to the 'metricSuccessfulRequestLatencyForOperations' call.");
+    }
+
+    const valueMetricNameMapper = (op: Operation) => `${op}Value`.toLowerCase();
+    const countMetricNameMapper = (op: Operation) => `${op}Count`.toLowerCase();
+
+    const values = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, props, valueMetricNameMapper);
+    const counts = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, { ...props, statistic: 'n' }, countMetricNameMapper);
+
+    const denominator = props.operations.map(op => `${countMetricNameMapper(op)}`).join(' + ');;
+    const numerator = props.operations.map(op => `${valueMetricNameMapper(op)} * ${countMetricNameMapper(op)}`).join(' + ');
+
+    return new cloudwatch.MathExpression({
+      expression: `(${numerator}) / (${denominator})`,
```

Comment:
I still have my doubts about this construction. I think it's bad practice.

First of all:

* I don't see the point of averaging the latency of `GetItem` requests (expected time <10ms) with the latency of `Scan` requests (expected time: ~couple 100ms). What kind of Alarm would you configure in a real-life scenario?
* I'm not even sure if taking an average of averages makes mathematical sense, but I'm 100% sure that in case the user configures a percentile statistic an average of percentiles makes absolutely no sense.

It might make sense to put all the latencies side-by-side in a single dashboard graph (something this method doesn't help with right now), but I'm not sure it makes sense to collapse all these latencies into a single metric.

Reply:
@rix0rrr This method can be used with a single operation, in which case it reduces to just the value of the individual metric. I agree combining `Scan` and `GetItem` doesn't make much sense, but what about `Scan` and `Query`? 

In some cases it might not exactly provide something meaningful, but in some cases it might, the user has all the power. Should i maybe just rename to `averageLatencyForOperation`? 

I don't mind getting rid of it - you think it does more harm than good?

Reply:
Removed this metric.